### ISSUE TITLE
using driftctl with GOOGLE_CREDENTIALS env var scans by default AWS with no warnings

### ISSUE THREAD

**Issue by sjourdan** [cloudskiff/driftctl#1018]
**Description**

- AWS is the default IaaS in driftctl. 
- So when an AWS_PROFILE is set etc., driftctl works well by default 
- But when explicitly asking for a GCP run with an appropriate env var, the developer UX is weird, because in this case where the user explicitly uses GOOGLE_CREDENTIALS, we can safely assume the user wanted to authenticate against GCP and not use AWS:

```sh
❯  GOOGLE_CREDENTIALS=keys/driftctl.json driftctl-0.15-pre scan 
[...]
DEBU[0000] Found existing provider path=/home/sjourdan/.driftctl/plugins/linux_amd64/terraform-provider-aws_v3.19.0_x5
DEBU[0002] Terraform provider initialized alias=us-east-1 name=aws
DEBU[0002] Found IAC supplier backend= path=terraform.tfstate supplier=tfstate
INFO[0002] Start reading IaC
DEBU[0002] Enumerated keys keys=[terraform.tfstate]
DEBU[0002] Reading resources from state backend= path=terraform.tfstate
DEBU[0002] Found module in state module= resourceCount=2
DEBU[0002] Unsupported provider found in state providerKey=google
[...]
```

**Environment**
 - OS: Fedora 34
 - driftctl version: 0.15-pre
 - terraform version: 1.0.5
 - terraform providers versions: google@3.82

**How to reproduce**

- Confirm you have a valid AWS driftctl scan env: 

```sh
$ env | grep AWS
AWS_PROFILE=driftctl
AWS_DEFAULT_REGION=us-east-1
AWS_REGION=us-east-1
AWS_HOME=/home/sjourdan/.aws
```

Run driftctl for GCP: 

```sh
$ cd some_tf_gcp_project
$ GOOGLE_CREDENTIALS=keys/driftctl.json driftctl-0.15-pre scan
[...]
Found resources not covered by IaC:
  aws_db_subnet_group:
    - default
[...]
```

**Possible Solution**

I'd expect a UX like: 

```sh 
❯  GOOGLE_CREDENTIALS=keys/driftctl.json driftctl-0.15-pre scan 
Error: driftctl is run with GCP environment variables but about to scan AWS.
Use driftctl scan --to gcp+tf to enable Terraform drifts detection on Google Cloud Platform (GCP).
```

We also know from the state that the provider is Google and not AWS. 

```json
{
  "version": 4,
  "terraform_version": "1.0.5",
  "serial": 9,
  "lineage": "8d177e17-e458-c76c-81b5-3a1b4979ffba",
  "outputs": {},
  "resources": [
    {
       [...]
      "provider": "provider[\"registry.terraform.io/hashicorp/google\"]",
      [... ]
```

When the execution environment has both AWS and GCP env variables, we can keep the default behavior for now because a single state can have multiple providers and we won't know which one to choose. 

Just help the user at this stage

**Additional context**

**Comment by sjourdan** [comment]
maybe to sync with https://github.com/cloudskiff/driftctl-docs/issues/105

(still, behavior is the same if using the currently valid `GOOGLE_APPLICATION_CREDENTIALS` var)

**Comment by sjourdan** [comment]
note: explicitly ask for `GOOGLE_CREDENTIALS, GOOGLE_CLOUD_KEYFILE_JSON or GCLOUD_KEYFILE_JSON` support

**Comment by craigfurman** [comment]
@sjourdan I agree with the use-case expressed in this issue, but I wonder if we can make it more general.

I'm assuming we only care about the single "credentialed" provider per module use-case, for now. i.e., a root module we're testing for drift would only have one of the AWS, GCP, or Azure providers, and any additional providers would be things like "random". Holler if this is wrong, and ignore the rest of this comment - it would need to be adapted.

In this case, I wonder if the ideal driftctl flow would be:

1. User runs `terraform init`
1. Driftctl looks for backend configuration in the local `.terraform` dir
2. Driftctl automatically uses credentials from the environment, if present, to pull the backend state
3. Driftctl infers what cloud account/project to scan based on the configured provider in the state that has been pulled from the backend.
4. Driftctl automatically uses credentials from the environment, if present, to scan the relevant cloud account.

How far are we from that now? The reason I bring all this up is that if this is indeed desirable, this issue could kind of be replaced by an issue that represents step 4, with the difference being that we wouldn't infer the cloud account from the environment - we'd infer it from the terraform state. WDYT?
